network/src/extension.rs

```diff
@@ -65,4 +65,6 @@ pub trait Extension: TimeoutHandler + Send + Sync {
     fn on_node_removed(&self, _node: &NodeId) {}
 
     fn on_message(&self, _node: &NodeId, _message: &[u8]) {}
+
+    fn on_timeout(&self, _token: TimerToken) {}
```

Comment:
I think we could remove default impl for `TimeoutHandler`

Reply:
ditto
https://github.com/CodeChain-io/codechain/pull/1353/commits/c0aa454844ec2f72f75d367689b467d5fe1aa842